AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -148,10 +146,12 @@ actor EndpointClient: EndpointClientBehaviour {
         }
 
         // 2. Look for a valid PinpointEndpointProfile object stored in the Keychain. It needs to match the current applicationId, otherwise we'll discard it.
-        if let endpointProfileData = try? keychain.getData(Constants.endpointProfileKey),
-           let decodedEndpointProfile = try? archiver.decode(PinpointEndpointProfile.self, from: endpointProfileData),
+        if let endpointProfileData = Self.getStoredData(from: keychain, forKey: Constants.endpointProfileKey, fallbackTo: userDefaults),
+            let decodedEndpointProfile = try? archiver.decode(PinpointEndpointProfile.self, from: endpointProfileData),
            decodedEndpointProfile.applicationId == configuration.appId {
             return await configure(endpointProfile: decodedEndpointProfile)
+        } else {
```

Comment:
**very** nit*ty*: Having the removal inside an `else` clause it not really needed, as the `return` on the happy path will break the execution.

Reply:
removed the `else` clause